### ISSUE TITLE
Functions for Calculating Skewness and Kurtosis 

### ISSUE THREAD

**Issue by Potato-Cracker** [pytorch/pytorch#101334]
### 🚀 The feature, motivation and pitch

I was working on a ML project recently involving statistical descriptors. Usually, I try to stick to torch functions(e.g- torch.mean() etc) but I was unable to find anything for skewness and kurtosis. I was looking for something similar to scipy.stats skew and kurtosis functions.

### Alternatives

For now, I have used scipy.stats.skew() and scipy.stats.kurtosis()

**Comment by DuanBoomer** [comment]
Wow that is a really nice catch. Please take a look at this.
> cc @svenstaro @JackDanger @infil00p @eklitzke

**Comment by reeshipaul** [comment]
i would like to work on this issue, please assign this to me

**Comment by lezcano** [comment]
I don't think we can implement this efficiently without a significant amount of effort in C++ / CUDA.

What about using the formulas in https://discuss.pytorch.org/t/statistics-for-whole-dataset/74511/2? You can then use `torch.compile` to compile them into something reasonably fast.

**Comment by DuanBoomer** [comment]
Well it sure would be really helpful to at least add these formulas to the documentation as a temporary solutions. In the meantime we can work on the actual builds for these formulas in C++ / CUDA.
Shall I start on this ?
cc @lezcano

**Comment by lezcano** [comment]
My point here is not about whether this could be done, but whether we should do it. These operations are fairly niche, and can be implemented in a few lines of code as discussed in https://discuss.pytorch.org/t/statistics-for-whole-dataset/74511/2. I do not know whether we really want to have them in core given how niche they are.

If we add them to core, we probably would simply add Python implementations of them, as no one is going to implement (and maintain!) their batching rules, forward and backward derivatives, AMP, JIT, `torch.compile` decomposition, meta function, etc. As such, it's not clear to me what are the benefits of adding these to core vs. having these and other tools implemented in an external librarly that specialises on statistical analysis (I'm sure there are already a few out there).
